### PR TITLE
21537: Adds smape as a prediction stat, MINOR

### PR DESCRIPTION
**Pull request by jackx111**
Adds smape as an available prediction stat to retrieve.

### CODE REVIEW

File: howso/feature_residuals.amlg [hunk 1]
```diff
@@ -621,6 +625,121 @@
...
+				(zip
+					numeric_continuous_features
+					;iterate over continuous features, where (current_value) is a list of react tuples for each feature
+					(map
+						(lambda
+							(if (size (current_value))
+								(let (assoc
+									adjusted_min_gap
+										(/
+											(if (contains_index !editDistanceFeatureTypesMap (get numeric_continuous_features (current_index 1)) )
```
Comment (cademack) [inline howso/feature_residuals.amlg]: can a "numeric_continuous" feature ever be an editDistance feature?

**Review by cademack — DISMISSED** [review]

File: howso/feature_residuals.amlg (hunk 1 above)
Comment (jackx111) [reply howso/feature_residuals.amlg]: actually don't think so

File: howso/feature_residuals.amlg [hunk 2]
```diff
@@ -621,6 +625,118 @@
...
+												(lambda
+													(let
+														(assoc
+															smape_numerator
+																(abs (-
+																	(get (current_value 1) predicted_offset)
+																	(get (current_value 1) actual_offset)
+																))
+															smape_denominator
+																(/ (+
```
Comment (howsoRes) [inline howso/feature_residuals.amlg]: all closing and opening parents should line up and be on same corresponding lines so please move the (+ to the next line.

File: howso/feature_residuals.amlg [hunk 3]
```diff
@@ -621,6 +625,118 @@
...
+				(zip
+					numeric_continuous_features
+					;iterate over continuous features, where (current_value) is a list of react tuples for each feature
+					(map
+						(lambda
+							(if (size (current_value))
+								(let (assoc
+									adjusted_min_gap
+										(/
+											(compute_on_contained_entities (list
```
Comment (howsoRes) [inline howso/feature_residuals.amlg]: not a fan of computing this everytime this is run. this should be cached somewhere during analyze. perhaps in the expectedValues map along with other counts?

**Review by howsoRes — APPROVED** [review]

**Review by howsohazard — APPROVED** [review]

**Pull request merged**